rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/utils.hpp

```diff
@@ -126,6 +126,21 @@ create_datareader(
   CustomDataReaderListener * listener,
   eprosima::fastdds::dds::DataReader ** data_reader);
 
+/**
+* Apply specific resource limits when using keys.
+* Max samples per instance is set to history depth if KEEP_LAST
+* else UNLIMITED.
+*
+* \param[in]       history_qos      History entitiy QoS.
+* \param[in, out]  res_limits_qos   Resource limits entitiy QoS.
```

Comment:
Is this correct? i think the `[input]` is the correct one.

```suggestion
* \param[inout]  res_limits_qos   Resource limits entitiy QoS.
```

Reply:
According to [Doxygen docs](https://www.doxygen.nl/manual/commands.html#cmdparam): _When a parameter is both input and output, [in,out] is used as attribute_